src/main/java/org/javarosa/xpath/expr/XPathEncryptStringFunc.java

```diff
@@ -40,13 +41,14 @@ private static String encryptString(Object o1, Object o2, Object o3) {
         String key = FunctionUtils.toString(o2);
         String algorithm = FunctionUtils.toString(o3);
 
-        if (!algorithm.equals("AES")) {
+        if (!algorithm.equals(getEncryptionKeyProvider().getAESKeyAlgorithmRepresentation()) &&
```

Comment:
See my comment [here](https://github.com/dimagi/commcare-core/pull/1349#discussion_r1425036234), I'm actually happy to drop this requirement it will make our implementation less complicated. I will wait for you thumbs up to proceed.

Reply:
Your comment above that makes sense to me although think we need to define our own `AES` and `RSA` constants here in context of the xpath functions as our Xpath syntax is not going to change based on what an encryption provider representation is.

Reply:
That's true, let me do that.